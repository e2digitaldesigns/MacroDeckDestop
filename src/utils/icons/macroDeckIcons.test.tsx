import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import MacroDeckIcon, { IntMacroDeckIcon } from "./macroDeckIcons";

const defaultProps: IntMacroDeckIcon = {
  icon: "home"
};

const setup = (props: IntMacroDeckIcon = defaultProps) => {
  return render(
    <MacroDeckIcon color={props.color} icon={props.icon} size={props.size} />
  );
};

describe("<MacroDeckIcon/>", () => {
  it("Should render without errors", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });

  it("Should show new attributes", () => {
    const props: IntMacroDeckIcon = {
      color: "orange",
      icon: "Activity",
      size: "36"
    };
    const component = setup(props);
    const svg = component.queryByTestId(`icon-${props.icon}`);
    expect(svg).toBeInTheDocument();
    expect(svg).toHaveAttribute("stroke", props.color);
    expect(svg).toHaveAttribute("width", props.size);
  });

  it("Should show home icon (default)", () => {
    const props: IntMacroDeckIcon = {
      icon: "not-a-real-icon"
    };
    const component = setup(props);
    const svg = component.getByTestId(`icon-Home`);
    expect(svg).toBeTruthy();
  });

  it("Should show letter icon)", () => {
    const props: IntMacroDeckIcon = {
      icon: "md-icon-alpha-B"
    };
    const component = setup(props);
    const svg = component.getByTestId(`icon-md-icon-alpha-B`);
    expect(svg).toBeTruthy();
  });
});
