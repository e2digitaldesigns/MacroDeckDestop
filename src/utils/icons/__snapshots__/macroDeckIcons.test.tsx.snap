// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MacroDeckIcon/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        data-testid="icon-Home"
        fill="none"
        height="24"
        stroke="white"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
        />
        <polyline
          points="9 22 9 12 15 12 15 22"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      data-testid="icon-Home"
      fill="none"
      height="24"
      stroke="white"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
      />
      <polyline
        points="9 22 9 12 15 12 15 22"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
